### PR TITLE
Fix loading data from configmaps.

### PR DESCRIPTION
**Pull request by dylanzr**
`ForAll` is looping through all the configmaps and decoding the content in to the `vol` struct, with how this code is, every one of them will point to the same struct in memory. As such, if you restart the provisioner pod, only the data from the last configmap loaded gets used.

### CODE REVIEW

File: Gopkg.lock [hunk 1]
```diff
@@ -528,6 +528,7 @@
     "k8s.io/apimachinery/pkg/api/errors",
     "k8s.io/apimachinery/pkg/apis/meta/v1",
     "k8s.io/apimachinery/pkg/util/sets",
+    "k8s.io/apimachinery/pkg/util/validation",
```
Comment (rootfs) [inline Gopkg.lock]: is it used?

**Comment by rootfs** [comment]
cc @mickymiek

File: Gopkg.lock (hunk 1 above)
Comment (dylanzr) [reply Gopkg.lock]: I actually don't think it is. I could remove it if you like. I think it would just get overwritten anyway.

**Comment by ShyamsundarR** [comment]
@dylanzr  request a port of the changes to master branch as well or, let me know and I can ensure the same. Thanks!

**Comment by humblec** [comment]
>@dylanzr request a port of the changes to master branch as well or, let me know and I can ensure the same. Thanks!

This is taken care by https://github.com/ceph/ceph-csi/pull/363

**Pull request merged**